compiler/src/iree/compiler/Dialect/Encoding/IR/EncodingAttrs.cpp

```diff
@@ -65,26 +66,225 @@ static FailureOr<AffineMap> getComposedAffineMap(Attribute attr) {
+                      "which is to be a `IntegerAttr`";
+               return failure();
+             }
+             return success();
+           },
+           [&]() {
+             opType = FieldParser<EncodingOpTypeAttr>::parse(p);
+             if (failed(opType)) {
+               p.emitError(p.getCurrentLocation())
+                   << "failed to parse EncodingAttr parameter 'op_type' "
+                      "which is to be a `EncodingOpTypeAttr`";
+               return failure();
+             }
+             return success();
+           },
+           [&]() {
+             elementTypes = FieldParser<ArrayAttr>::parse(p);
+             if (failed(elementTypes)) {
+               p.emitError(p.getCurrentLocation())
+                   << "failed to parse EncodingAttr parameter 'element_types' "
+                      "which is to be a `ArrayAttr`";
+               return failure();
+             }
+             return success();
+           },
+           [&]() {
+             userIndexingMaps = FieldParser<ArrayAttr>::parse(p);
+             if (failed(userIndexingMaps)) {
+               p.emitError(p.getCurrentLocation())
+                   << "failed to parse EncodingAttr parameter "
+                      "'user_indexing_maps' which is to be a `ArrayAttr`";
+               return failure();
+             }
+             return success();
+           },
+           [&]() {
+             iterationSizes = parseDynamicI64ArrayAttr(p);
+             if (failed(iterationSizes)) {
+               p.emitError(p.getCurrentLocation())
+                   << "failed to parse EncodingAttr parameter "
+                      "'iteration_sizes' "
+                      "which is to be a `ArrayAttr`";
+               return failure();
+             }
+             return success();
+           },
+           [&]() {
+             layouts = FieldParser<ArrayAttr>::parse(p);
+             if (failed(layouts)) {
+               p.emitError(p.getCurrentLocation())
+                   << "failed to parse EncodingAttr parameter 'layouts' which "
+                      "is to be a `ArrayAttr`";
+               return failure();
+             }
+             return success();
+           }}))) {
+    p.emitError(p.getCurrentLocation()) << "failed parsing encoding attribute";
+    return {};
+  }
+  return p.getChecked<EncodingAttr>(
+      loc, p.getContext(), *operandIndex, *opType, *elementTypes,
+      userIndexingMaps.value_or(ArrayAttr()),
+      iterationSizes.value_or(ArrayAttr()), layouts.value_or(ArrayAttr()));
+}
+
+void EncodingAttr::print(AsmPrinter &p) const {
+  p << "<";
+  p << "operand_index = ";
+  p.printStrippedAttrOrType(getOperandIndex());
+  p << ", op_type = ";
+  p.printStrippedAttrOrType(getOpType());
+  p << ", element_types = ";
+  p.printStrippedAttrOrType(getElementTypes());
+  if (!(getUserIndexingMaps() == ArrayAttr())) {
+    p << ", user_indexing_maps = ";
+    p.printStrippedAttrOrType(getUserIndexingMaps());
+  }
```

Comment:
Can we declare the variable in the if condition? It is more common in MLIR/IREE codebase.

```suggestion
  if (auto maps = getUserIndexingMaps()) {
    p << ", user_indexing_maps = ";
    p.printStrippedAttrOrType(maps);
  }
```

Reply:
Thanks, this was copy paste from the tablegen. Updated it.